lib/cluster-providers/generic-cluster-provider.ts

```diff
@@ -0,0 +1,249 @@
+        const privateCluster = this.props.privateCluster ?? valueFromContext(scope, constants.PRIVATE_CLUSTER, false);
+        const endpointAccess = (privateCluster === true) ? eks.EndpointAccess.PRIVATE : eks.EndpointAccess.PUBLIC_AND_PRIVATE;
+        const vpcSubnets = this.props.vpcSubnets ?? (privateCluster === true) ? [{ subnetType: ec2.SubnetType.PRIVATE_WITH_NAT }] : undefined;
+
+        const defaultOptions = {
+            vpc,
+            clusterName,
+            outputClusterName,
+            version,
+            vpcSubnets,
+            endpointAccess,
+            defaultCapacity: 0 // we want to manage capacity ourselves
+        };
+
+        const clusterOptions = {...this.props, ...defaultOptions };
+        // Create an EKS Cluster
+        const cluster = this.internalCreateCluster(scope, id, clusterOptions);
+
+        const nodeGroups: eks.Nodegroup[] = [];
+        
+        this.props.managedNodeGroups?.forEach( n => {
+            const nodeGroup = this.addManagedNodeGroup(cluster, n);
+            nodeGroups.push(nodeGroup);
+        });
+
+        const autoscalingGroups: autoscaling.AutoScalingGroup[] = [];
+        this.props.autoscalingNodeGroups?.forEach( n => {
+            const autoscalingGroup = this.addAutoScalingGroup(cluster, n);
+            autoscalingGroups.push(autoscalingGroup);
+        });
+
+        const fargateProfiles = Object.entries(this.props.fargateProfiles ?? {});
+        fargateProfiles?.forEach(([key, options]) => this.addFargateProfile(cluster, key, options));
+
+        return new ClusterInfo(cluster, version, nodeGroups, autoscalingGroups);
+    }
+
+    /**
+     * Template method that may be overridden by subclasses to create a specific cluster flavor (e.g. FargateCluster vs eks.Cluster)
+     * @param scope 
+     * @param id 
+     * @param clusterOptions 
+     * @returns 
+     */
+    protected internalCreateCluster(scope: Construct, id: string, clusterOptions: any) : eks.Cluster {
+        return new eks.Cluster(scope, id, clusterOptions);
+    }
+
+    /**
+     * Adds an autoscaling group to the cluster.
+     * @param cluster 
+     * @param nodeGroup 
+     * @returns 
+     */
+    addAutoScalingGroup(cluster: eks.Cluster, nodeGroup: AutoscalingNodeGroup): autoscaling.AutoScalingGroup {
+        const machineImageType = nodeGroup.machineImageType ?? eks.MachineImageType.AMAZON_LINUX_2;
+        const instanceType = nodeGroup.instanceType ?? valueFromContext(cluster, constants.INSTANCE_TYPE_KEY, constants.DEFAULT_INSTANCE_TYPE);
+        const minSize = nodeGroup.minSize ?? valueFromContext(cluster, constants.MIN_SIZE_KEY, constants.DEFAULT_NG_MINSIZE);
+        const maxSize = nodeGroup.maxSize ?? valueFromContext(cluster, constants.MAX_SIZE_KEY, constants.DEFAULT_NG_MAXSIZE);
+        const desiredSize = nodeGroup.desiredSize ?? valueFromContext(cluster, constants.DESIRED_SIZE_KEY, minSize);
+        const updatePolicy = UpdatePolicy.rollingUpdate();
```

Comment:
Can we configure batch size for rolling update ?

Reply:
great idea.

Reply:
@allamand  This option is unfortunately not exposed through CDK at the moment, here is what they have [available](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_eks.NodegroupOptions.html). It is surely a miss. I will create an issue for that for CDK or i will move this to a lower level construct that will use direct EKS API. Thanks for bringing it up.

Reply:
addressed for ASG in the last commit, you can pass your update policy, set replace or rolling with options. 
For managed node groups, however, there is no direct support in CDK (including v2), other than falling back to lower level (CFN) constructs. I created https://github.com/aws-quickstart/ssp-amazon-eks/issues/309 to address.